Detectors/ZDC/calib/src/WaveformCalibEPNSpec.cxx

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   WaveformCalibEPNSpec.cxx
+/// @brief  ZDC reconstruction
+/// @author pietro.cortese@cern.ch
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "CCDB/CcdbApi.h"
+#include "Framework/Logger.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/CCDBParamSpec.h"
+#include "Framework/DataRefUtils.h"
+#include "DetectorsCommonDataFormats/DetID.h"
+#include "DataFormatsZDC/BCData.h"
+#include "DataFormatsZDC/ChannelData.h"
+#include "DataFormatsZDC/OrbitData.h"
+#include "DataFormatsZDC/RecEvent.h"
+#include "ZDCBase/ModuleConfig.h"
+#include "CommonUtils/NameConf.h"
+#include "CommonUtils/MemFileHelper.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "ZDCCalib/WaveformCalibData.h"
+#include "ZDCCalib/WaveformCalibEPNSpec.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace zdc
+{
+
+WaveformCalibEPNSpec::WaveformCalibEPNSpec()
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+WaveformCalibEPNSpec::WaveformCalibEPNSpec(const int verbosity) : mVerbosity(verbosity)
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+void WaveformCalibEPNSpec::init(o2::framework::InitContext& ic)
+{
+  mVerbosity = ic.options().get<int>("verbosity-level");
+  mWorker.setVerbosity(mVerbosity);
+  mTimer.CpuTime();
+  mTimer.Start(false);
```

Comment:
If you want to measure the time when your code was actually running, you should stop the timer here, not start.
And then, you should `Start(false)` it at the entrance to run() and `Stop()` on exit from it.